src/bin/cargo/commands/install.rs

```diff
@@ -108,6 +109,18 @@ pub fn exec(config: &mut Config, args: &ArgMatches) -> CliResult {
         .map(|k| resolve_crate(k, version))
         .collect::<crate::CargoResult<Vec<_>>>()?;
 
+    for krate in krates.iter() {
+        let crate_name = krate.0;
+        if !crate_name.starts_with("+") {
+            continue;
+        }
+
+        let crate_name_wo_prefix = crate_name[1..].to_string();
```

Comment:
Unnecessary `to_string` allocation.

```suggestion
        let crate_name_wo_prefix = &crate_name[1..];
```